ecl/eclccserver/eclccserver.cpp

```diff
@@ -301,7 +342,43 @@ class EclccCompileThread : implements IPooledThread, implements IErrorReporter,
         }
     }
 
-    unsigned executeLine(const char *line, StringBuffer &output, bool alreadyFailed)
+    unsigned doRunCompileCommand(AbortWaiter &abortWaiter, StringBuffer &output, const char *cmd)
+    {
+        try
+        {
+            Owned<IPipeProcess> pipe = createPipeProcess();
+            AbortPipeWaiter aborter(abortWaiter, pipe);
+            int ret = START_FAILURE;
```

Comment:
Should this check abortWaiter.hasTimedout()? otherwise there is a small race condition.  Or better recode IPipeProcess to save that it has been aborted, and fail when run is being called (otherwise there will still be a window).